dist/aframe-altspace-component.js

```diff
@@ -465,86 +467,744 @@
+				});
+
+				this.clientsRef.on('child_added', function(childSnapshot) {
+					var joinedClientId = childSnapshot.val();
+					//let the master client flag get set first
+					setTimeout(function(){
+						component.sceneEl.emit('clientjoined', {id: joinedClientId}, false);
+					}, 1);
+				});
+
+				this.clientsRef.on('child_removed', function(childSnapshot) {
+					var leftClientId = childSnapshot.val();
+					//let the master client flag get set first
+					setTimeout(function(){
+						component.sceneEl.emit('clientleft', {id: leftClientId}, false);
+					}, 1);
+				});
+
+				// add our client ID to the list of connected clients, 
+				// but have it be automatically removed by firebase if we disconnect for any reason
+				this.clientsRef.push(this.clientId).onDisconnect().remove();
+
+				this.connection.instance.child('initialized').once('value', function (snapshot) {
+					var shouldInitialize = !snapshot.val();
+					snapshot.ref().set(true);
+
+					this.sceneEl.emit('instanceready', { shouldInitialize: shouldInitialize }, false);
+				}.bind(this));
+
+
+				Object.defineProperty(this, 'isMasterClient', {
+					get: function () { return masterClientId === this.clientId; }.bind(this)
+				});
+			}.bind(this));
+		}
+	});
+
+	AFRAME.registerComponent('sync',
+	{
+		schema: {
+			mode: { default: 'link' }
+		},
+		init: function () {
+			var scene = document.querySelector('a-scene');
+			var syncSys = scene.systems['sync-system'];
+
+			var ref;
+			var key;
+			var dataRef;
+			var ownerRef;
+			var ownerId;
+			var isMine = false;
+
+			var component = this;
+
+			//Make sure someone always owns an object. If the owner leaves and we are the master client, we will take it.
+			//This ensures, for example, that synced animations keep playing
+			scene.addEventListener('clientleft', function(event){
+				var shouldTakeOwnership = (!ownerId || ownerId === event.detail.id) && syncSys.isMasterClient;
+
+				if(shouldTakeOwnership) component.takeOwnership();
+			});
+
+			if (this.data.mode === 'link') {
+				var id = this.el.id;
+				if (!id) {
+					console.error('Entities cannot be synced using link mode without an id.');
+					return;
+				}
+
+				link(syncSys.sceneRef.child(id));
+				setupReceive();
+			}
+
+			function link(entityRef) {
+				ref = entityRef;
+				key = ref.key();
+				dataRef = ref.child('data');
+				component.dataRef = dataRef;
+				ownerRef = ref.child('owner');
+			}
+
+			function setupReceive() {
+
+				//if nobody has owned the object yet, we will. 
+				ownerRef.transaction(function (owner) {
+					if (owner) return undefined;
+
+					ownerRef.onDisconnect().set(null);
+					return syncSys.clientId;
+				});
+
+				ownerRef.on('value',
+					function(snapshot) {
+						var newOwnerId = snapshot.val();
+
+						var gained = newOwnerId === syncSys.clientId && !isMine;
+						if (gained) component.el.emit('ownershipgained', null, false);
+
+
+						//note this also fires when we start up without ownership
```

Comment:
Yeah.. not sure what the intended behavior should be. It makes the transform sync code simpler if it does the above behavior.